accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -0,0 +1,1212 @@
+the case where the values of the type variables do not satisfy their
+declared bounds, and those values will be obtained directly from the static
+type of the receiver in each member invocation on `V`.*
+
+A compile-time error occurs if a type parameter of an extension type
+declaration occurs in a non-covariant position in the representation type.
+
+When `s` is zero,
+<code>V&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>
+simply stands for `V`, a non-generic extension type.
+When `s` is greater than zero, a raw occurrence `V` is treated like a raw
+type: Instantiation to bound is used to obtain the omitted type arguments.
+*Note that this may yield a super-bounded type, which is then a
+compile-time error.*
+
+We say that the static type of said variable, parameter, etc.
+_is the extension type_
+<code>V&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>,
+and that its static type _is an extension type_.
+
+It is a compile-time error if `await e` occurs, and the static type of
+`e` is an extension type which is not a subtype of `Future<T>` or
+`FutureOr<T>` for any `T`.
+
+A compile-time error occurs if an extension type declares a member whose
+name is declared by `Object` as well.
+
+*For example, an extension type declaration cannot declare an operator `==`
+or a member named `noSuchMethod` or `toString`. The rationale is that these
+extension type methods would be highly confusing and error prone. For
+example, collections would still call the instance operator `==`, not the
+extension type operator, and string interpolation would call the instance
+method `toString`, not the extension type method. Also, when we make this an
+error for now, we have the option to allow it, perhaps with some
+restrictions, in a future version of Dart.*
+
+A compile-time error occurs if an extension type is used as a
+superinterface of a class or a mixin, or if an extension type is used to
+derive a mixin.
+
+*In other words, an extension type cannot occur as a superinterface in an
+`extends`, `with`, `implements`, or `on` clause of a class or mixin.  On
+the other hand, it can occur in other ways, e.g., as a type argument of a
+superinterface of a class.*
+
+If `e` is an expression whose static type `V` is the extension type
+<code>Name&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code> and `m` is the
+name of a member that `V` has, a member access like `e.m(args)` is treated
+as an invocation of the extension type member `m` on the receiver `e`
+according to the extension type declaration `Name` and with the actual type
+arguments <code>T<sub>1</sub>, ..., T<sub>s</sub></code>, with the actual
+argument part `args`.
+
+Similarly, `e.m` is treated an invocation of the extension type member `m`
+on the receiver `e` according to the extension type declaration `Name` and
+with the actual type arguments <code>T<sub>1</sub>, ...,
+T<sub>s</sub></code> and no actual argument part.
+
+*Setter invocations are treated as invocations of methods with a
+single argument.*
```

Comment:
And operators are treated like funny-named methods, `e1 + e2` works as if `+` was a method name and was invoked as `e1.+(e2)`.

Reply:
Done.